test/integration/dnspolicy_controller_test.go

```diff
@@ -327,8 +333,7 @@ var _ = Describe("DNSPolicy", Ordered, func() {
 					// must exist
 					Expect(err).ToNot(HaveOccurred())
 					return existingGateway.GetAnnotations()
-				}, time.Second*5, time.Second).ShouldNot(HaveKey(DNSPolicyBackRefAnnotation))
-
+				}, time.Second*5, time.Second).ShouldNot(HaveKeyWithValue(DNSPoliciesBackRefAnnotation, policiesBackRefValue))
```

Comment:
Is this because the annotation still exists, but with a different value?

Reply:
There are 2 annotations we need to check and it was incorrectly checking the same one twice. The `DNSPoliciesBackRefAnnotation` won't be removed completely on deletion, but should be empty.